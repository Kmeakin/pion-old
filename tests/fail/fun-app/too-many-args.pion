//~ run = "${PION} elab ${FILE}"

{
let id = fun (x : Bool) => x;
let _ = id(false, true);
let _ = id(false, true, true);

let always = fun (x : Bool) (y : Int) => y;
let _ = always(false, 0, true);
let _ = always(false, 0, true, true);

Type
}
