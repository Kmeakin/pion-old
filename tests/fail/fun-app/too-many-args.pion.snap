exit status: 0
stdout:
let id : Bool -> Bool = fun (x : Bool) => x; let _ : #error = #error;
let _ : #error = #error;
let always : Bool -> Int -> Int = fun (x : Bool) (y : Int) => y;
let _ : #error = #error; let _ : #error = #error; Type : Type

stderr:
error: tried to apply too many arguments to function
  ┌─ tests/fail/fun-app/too-many-args.pion:4:9
  │
4 │ let _ = id false true;
  │         ^^       ---- extra argument
  │         │         
  │         expression of type `Bool -> Bool`
  │
  = help: function expects 1 argument, but recieved 2 arguments

error: tried to apply too many arguments to function
  ┌─ tests/fail/fun-app/too-many-args.pion:5:9
  │
5 │ let _ = id false true true;
  │         ^^       --------- extra arguments
  │         │         
  │         expression of type `Bool -> Bool`
  │
  = help: function expects 1 argument, but recieved 3 arguments

error: tried to apply too many arguments to function
  ┌─ tests/fail/fun-app/too-many-args.pion:8:9
  │
8 │ let _ = always false 0 true;
  │         ^^^^^^         ---- extra argument
  │         │               
  │         expression of type `Bool -> Int -> Int`
  │
  = help: function expects 2 arguments, but recieved 3 arguments

error: tried to apply too many arguments to function
  ┌─ tests/fail/fun-app/too-many-args.pion:9:9
  │
9 │ let _ = always false 0 true true;
  │         ^^^^^^         --------- extra arguments
  │         │               
  │         expression of type `Bool -> Int -> Int`
  │
  = help: function expects 2 arguments, but recieved 4 arguments

